src/module.cc

```diff
@@ -0,0 +1,338 @@
+    isolate->SetAbortOnUncaughtExceptionCallback(OnUncaughtException);
+  }
+
+#ifndef _WIN32
+  // If NodeReport newly requested on external user signal set up watchdog thread and callbacks
+  if ((nodereport_events & NR_SIGNAL) && !(previous_events & NR_SIGNAL)) {
+    uv_sem_init(&report_semaphore, 0);
+    if (StartWatchdogThread(ReportSignalThreadMain) == 0) {
+      uv_async_init(uv_default_loop(), &nodereport_trigger_async, SignalDumpAsyncCallback);
+      uv_unref(reinterpret_cast<uv_handle_t*>(&nodereport_trigger_async));
+      RegisterSignalHandler(nodereport_signal, SignalDump, false);
+    }
+  }
+#endif
+}
+NAN_METHOD(SetCoreDump) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  nodereport_events = ProcessNodeReportCoreSwitch(*parameter);
+}
+NAN_METHOD(SetSignal) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  nodereport_events = ProcessNodeReportSignal(*parameter);
+}
+NAN_METHOD(SetFileName) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  ProcessNodeReportFileName(*parameter);
+}
+NAN_METHOD(SetDirectory) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  ProcessNodeReportDirectory(*parameter);
+}
+NAN_METHOD(SetVerbose) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  nodereport_verbose = ProcessNodeReportVerboseSwitch(*parameter);
+}
+
+/*******************************************************************************
+ * Callbacks for triggering NodeReport on failure events (as configured)
+ *  - fatal error
+ *  - uncaught exception
+ *  - signal
+ ******************************************************************************/
+static void OnFatalError(const char* location, const char* message) {
+  if (location) {
+    fprintf(stderr, "FATAL ERROR: %s %s\n", location, message);
+  } else {
+    fprintf(stderr, "FATAL ERROR: %s\n", message);
+  }
+  // Trigger NodeReport if requested
+  if (nodereport_events & NR_FATALERROR) {
+    TriggerNodeReport(Isolate::GetCurrent(), kFatalError, message, location, NULL);
+  }
+  fflush(stderr);
+  if (nodereport_core) {
+    raise(SIGABRT); // core dump requested
+  } else {
+    exit(0); // no core dump requested
+  }
+}
+
+bool OnUncaughtException(v8::Isolate* isolate) {
+   // Trigger NodeReport if required
+  if (nodereport_events & NR_EXCEPTION) {
+    TriggerNodeReport(Isolate::GetCurrent(), kException, "exception", "OnUncaughtException (nodereport/src/module.cc)", NULL);
+  } 
+  if (nodereport_core) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+#ifndef _WIN32
+static void SignalDumpInterruptCallback(Isolate *isolate, void *data) {
+  if (report_signal != 0) {
+    fprintf(stderr,"SignalDumpInterruptCallback - handling signal\n");
+    if (nodereport_events & NR_SIGNAL) {
+      if (nodereport_verbose) {
+        fprintf(stderr,"SignalDumpInterruptCallback - triggering NodeReport\n");
+      }
+      TriggerNodeReport(Isolate::GetCurrent(), kSignal_JS,

```

Comment:
You get the isolate as a parameter, right?  No need to call `Isolate::GetCurrent()`.